crates/matrix-sdk-base/src/rooms/mod.rs

```diff
@@ -569,4 +604,19 @@ mod tests {
         base_room_info.handle_notable_tags(&tags);
         assert!(base_room_info.notable_tags.contains(RoomNotableTags::LOW_PRIORITY).not());
     }
+
+    #[test]
+    fn test_get_user_id_from_state_key() {
+        assert_matches!(get_user_id_for_state_key(""), None);
+        assert_matches!(get_user_id_for_state_key("abc"), None);
+        assert_matches!(get_user_id_for_state_key("username:example.org"), None);
+
+        assert_matches!(get_user_id_for_state_key("@username:example.org"), Some(_));
```

Comment:
Can you explicit, in all of the `Some()` tests, what the expected value is? e.g.

```suggestion
        assert_matches!(get_user_id_for_state_key("@username:example.org"), Some(user_id!("@username:example.org"));
```

Reply:
[`699369a`](https://github.com/matrix-org/matrix-rust-sdk/pull/3836/commits/699369aa2876c98144ac220227e68e8e12094834) -- let me know if this too verbose.